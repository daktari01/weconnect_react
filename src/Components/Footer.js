import React, { Component } from 'react';
import ReactDOM from 'react-dom';

class Footer extends Component{
    render(){
        return(
            <footer className="footer">
                <div className="container">
                    <div className="row">
                        <div className="col-sm-12">
                            <span>WeConnect &copy;2018. All rights reserved. Designed and maintained by Daktari Technologies.</span>
                        </div>
                    </div>
                </div>
            </footer>
        );
    }
}

export default Footer;