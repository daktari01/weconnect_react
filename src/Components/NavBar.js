import React, { Component } from "react";
import axios from "axios";
import { Redirect } from "react-router-dom";
import logo from "../static/img/logos/weConnect.png";
import { toast } from "react-toastify";
import { localApi } from "../utilities/api";

class NavBar extends Component {
  state = {
    fireRedirect: false
  };
  componentWillMount() {
    this.setState({ logged_in: false });
  }
  componentDidMount() {
    const token = localStorage.getItem("access_token");
    if (token) {
      this.setState({ logged_in: true });
    }
  }
  handleLogout = event => {
    const access_token = localStorage.getItem("access_token");
    event.preventDefault();
    axios
      .post(
        localApi + "auth/logout",
        {},
        {
          headers: {
            Accept: "application/json",
            "Content-type": "application/json",
            "x-access-token": access_token
          }
        }
      )
      .then(response => {
        window.location.reload();
        toast.success(response.data["message"]);
        this.setState({ logged_in: false, fireRedirect: true });
        localStorage.removeItem("access_token");
        localStorage.removeItem("user_id");
        localStorage.removeItem("isAuthenticated");
        console.log(this.state.logged_in);
      })
      .catch(error => {
        toast.error("Log out unsuccessful. Please try again");
      });
  };
  render() {
    const { fireRedirect, logged_in } = this.state;
    return (
      <div className="container">
        <nav className="navbar navbar-expand-lg navbar-light bg-faded">
          <a className="navbar-brand" href="/">
            <img src={logo} className="logo" alt="WeConnect logo" />
          </a>
          <button
            className="navbar-toggler"
            type="button"
            data-toggle="collapse"
            data-target="#navbarNavDropdown"
            aria-controls="navbarNavDropdown"
            aria-expanded="false"
            aria-label="Toggle navigation"
          >
            <span className="navbar-toggler-icon" />
          </button>
          {logged_in ? (
            <div id="navbarNavDropdown" className="navbar-collapse collapse">
              <ul className="navbar-nav mr-auto">
                <li className="nav-item active">
                  <a className="nav-link" href="/businesses">
                    All businesses <span className="sr-only">(current)</span>
                  </a>
                </li>
                <li className="nav-item">
                  <a className="nav-link" href="/my-businesses">
                    My businesses <span className="sr-only">(current)</span>
                  </a>
                </li>
                <li className="nav-item">
                  <a className="nav-link" href="/register-business">
                    Register business <span className="sr-only">(current)</span>
                  </a>
                </li>
              </ul>
              <ul className="navbar-nav">
                <li className="nav-item">
                  <a className="nav-link" href="#" onClick={this.handleLogout}>
                    Logout
                  </a>
                </li>
              </ul>
            </div>
          ) : (
            <div id="navbarNavDropdown" className="navbar-collapse collapse">
              <ul className="navbar-nav ml-auto">
                <li className="nav-item">
                  <a className="nav-link" href="/login">
                    Login
                  </a>
                </li>
                <li className="nav-item">
                  <a className="nav-link" href="/register">
                    Register
                  </a>
                </li>
              </ul>
            </div>
          )}
        </nav>
        {fireRedirect && <Redirect to={"/login"} />}
      </div>
    );
  }
}

export default NavBar;
