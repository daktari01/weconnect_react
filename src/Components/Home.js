import React, { Component } from "react";
import { Redirect } from "react-router-dom";
import "react-toastify/dist/ReactToastify.css";
import NavBar from "./NavBar";
import Footer from "./Footer";
import {
  TabContent,
  TabPane,
  Nav,
  NavItem,
  NavLink,
  Row,
  Col
} from "reactstrap";
import classnames from "classnames";

/**
 * Handle the Home page
 */
class Home extends Component {
  constructor(props) {
    super(props);

    this.toggle = this.toggle.bind(this);
    this.state = {
      queryType: "q",
      fireRedirect: false,
      query: "",
      activeTab: "q"
    };
  }

  /**
   * Handles onchange event for search input boxes
   */
  handleInput = event => {
    this.setState({
      query: event.target.value
    });
  };

  /**
   * Handles the search buttons for the search
   */
  handleSearch = event => {
    event.preventDefault();
    this.setState({ fireRedirect: true });
  };

  /**
   * Handle the search box
   * @param tab
   */
  toggle(tab) {
    if (this.state.activeTab !== tab) {
      this.setState({
        activeTab: tab,
        queryType: tab
      });
    }
  }
  render() {
    const { fireRedirect } = this.state;
    return (
      <div>
        <section id="body">
          <NavBar />
          <div className="container">
            <div className="d-flex searchContainer" id="loginContainer">
              <div
                id="loginWrapper"
                className="justify-content-center align-self-center"
              >
                <div className="search-box white-bg" id="indexSearchBox">
                  <Nav tabs>
                    <NavItem>
                      <NavLink
                        className={classnames({
                          active: this.state.activeTab === "q"
                        })}
                        onClick={() => {
                          this.toggle("q");
                        }}
                      >
                        Business Name
                      </NavLink>
                    </NavItem>
                    <NavItem>
                      <NavLink
                        className={classnames({
                          active: this.state.activeTab === "location"
                        })}
                        onClick={() => {
                          this.toggle("location");
                        }}
                      >
                        Business Location
                      </NavLink>
                    </NavItem>
                    <NavItem>
                      <NavLink
                        className={classnames({
                          active: this.state.activeTab === "category"
                        })}
                        onClick={() => {
                          this.toggle("category");
                        }}
                      >
                        Business Category
                      </NavLink>
                    </NavItem>
                  </Nav>
                  <TabContent activeTab={this.state.activeTab}>
                    <TabPane tabId="q">
                      <Row>
                        <Col sm="12">
                          <div
                            className="tab-pane fade show active"
                            id="pills-name"
                            role="tabpanel"
                            aria-labelledby="pills-name-tab"
                          >
                            <div className="input-group mb-3">
                              <input
                                type="text"
                                className="form-control inputSearch"
                                placeholder="Search business by name ..."
                                aria-label="pills-name-input"
                                aria-describedby="basic-addon2"
                                name="inputName"
                                onChange={this.handleInput}
                              />
                              <div className="input-group-append">
                                <button
                                  className="btn btn-outline-secondary btnSearch"
                                  type="button"
                                  onClick={this.handleSearch}
                                >
                                  Search
                                </button>
                              </div>
                            </div>
                          </div>
                        </Col>
                      </Row>
                    </TabPane>
                    <TabPane tabId="location">
                      <Row>
                        <Col sm="12">
                          <div
                            className="tab-pane fade show active"
                            id="pills-name"
                            role="tabpanel"
                            aria-labelledby="pills-name-tab"
                          >
                            <div className="input-group mb-3">
                              <input
                                type="text"
                                className="form-control inputSearch"
                                placeholder="Search business by location ..."
                                aria-label="pills-name-input"
                                aria-describedby="basic-addon2"
                                name="inputLocation"
                                onChange={this.handleInput}
                              />
                              <div className="input-group-append">
                                <button
                                  className="btn btn-outline-secondary btnSearch"
                                  type="button"
                                  onClick={this.handleSearch}
                                >
                                  Search
                                </button>
                              </div>
                            </div>
                          </div>
                        </Col>
                      </Row>
                    </TabPane>
                    <TabPane tabId="category">
                      <Row>
                        <Col sm="12">
                          <div
                            className="tab-pane fade show active"
                            id="pills-name"
                            role="tabpanel"
                            aria-labelledby="pills-name-tab"
                          >
                            <div className="input-group mb-3">
                              <input
                                type="text"
                                className="form-control inputSearch"
                                placeholder="Search business by category ..."
                                aria-label="pills-name-input"
                                aria-describedby="basic-addon2"
                                name="inputCategory"
                                onChange={this.handleInput}
                              />
                              <div className="input-group-append">
                                <button
                                  className="btn btn-outline-secondary btnSearch"
                                  type="button"
                                  onClick={this.handleSearch}
                                >
                                  Search
                                </button>
                              </div>
                            </div>
                          </div>
                        </Col>
                      </Row>
                    </TabPane>
                  </TabContent>
                </div>
              </div>
            </div>
          </div>
        </section>
        {fireRedirect && (
          <Redirect to={{ pathname: "/businesses", searchProps: this.state }} />
        )}
        <Footer />
      </div>
    );
  }
}

export default Home;
