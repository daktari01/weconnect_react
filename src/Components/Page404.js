import React, { Component } from "react";
import NavBar from "./NavBar";
import Footer from "./Footer";
import signInLogo from "../static/img/logos/weConnect.png";
import chimp from "../static/img/backgrounds/chimp404.jpg";

class Page404 extends Component {
  render() {
    return (
      <div>
        <section id="body">
          <NavBar />
          <div className="container image404 col-min-6" id="loginContainer">
            <div className="wrapper">
              <div className="card">
                <div className="row">
                  <div className="col-sm-4">
                    <img
                      id="img404"
                      className="img-fluid"
                      alt="page404"
                      src={chimp}
                    />
                  </div>
                  <div className="col-sm-8">
                    <div id="text404" className="align-middle">
                    <h1>404</h1>
                    <h3>Page Not Found</h3>
                    <h4>The page you are looking for could be found on this server.</h4>
                    <br />
                    <a href="/" className="btn btn-primary">Take Me Home</a>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </section>
        <Footer />
      </div>
    );
  }
}

export default Page404;
