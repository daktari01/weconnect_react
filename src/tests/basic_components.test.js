import Enzyme, { shallow, render, mount } from "enzyme";
import React from "react";
import Footer from "../Components/Footer";
import Page404 from "../Components/Page404";
import NavBar from "../Components/NavBar";
import Home from "../Components/Home";
import { shallowToJson } from "enzyme-to-json";

describe("Footer component", () => {
  const wrapper = shallow(<Footer />);
  it("renders footer successfully", () => {
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
});

describe("Page404 renders correctly", () => {
  const wrapper = shallow(<Page404 />);
  it("Succeeds", () => {
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
});

describe("Home renders correctly", () => {
  const wrapper = shallow(<Home />);
  it("Succeeds", () => {
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
});

describe("NavBar renders correctly", () => {
  const wrapper = shallow(<NavBar />);
  it("Succeeds", () => {
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
});
