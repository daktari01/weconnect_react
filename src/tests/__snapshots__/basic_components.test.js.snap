// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer component renders footer successfully 1`] = `
<footer
  className="footer"
>
  <div
    className="container"
  >
    <div
      className="row"
    >
      <div
        className="col-sm-12"
      >
        <span>
          WeConnect Â©2018. All rights reserved. Designed and maintained by Daktari Technologies.
        </span>
      </div>
    </div>
  </div>
</footer>
`;

exports[`Home renders correctly Succeeds 1`] = `
<div>
  <section
    id="body"
  >
    <NavBar />
    <div
      className="container"
    >
      <div
        className="d-flex searchContainer"
        id="loginContainer"
      >
        <div
          className="justify-content-center align-self-center"
          id="loginWrapper"
        >
          <div
            className="search-box white-bg"
            id="indexSearchBox"
          >
            <Nav
              tabs={true}
              tag="ul"
              vertical={false}
            >
              <NavItem
                tag="li"
              >
                <NavLink
                  className="active"
                  onClick={[Function]}
                  tag="a"
                >
                  Business Name
                </NavLink>
              </NavItem>
              <NavItem
                tag="li"
              >
                <NavLink
                  className=""
                  onClick={[Function]}
                  tag="a"
                >
                  Business Location
                </NavLink>
              </NavItem>
              <NavItem
                tag="li"
              >
                <NavLink
                  className=""
                  onClick={[Function]}
                  tag="a"
                >
                  Business Category
                </NavLink>
              </NavItem>
            </Nav>
            <TabContent
              activeTab="q"
              tag="div"
            >
              <TabPane
                tabId="q"
                tag="div"
              >
                <Row
                  tag="div"
                >
                  <Col
                    sm="12"
                    tag="div"
                    widths={
                      Array [
                        "xs",
                        "sm",
                        "md",
                        "lg",
                        "xl",
                      ]
                    }
                  >
                    <div
                      aria-labelledby="pills-name-tab"
                      className="tab-pane fade show active"
                      id="pills-name"
                      role="tabpanel"
                    >
                      <div
                        className="input-group mb-3"
                      >
                        <input
                          aria-describedby="basic-addon2"
                          aria-label="pills-name-input"
                          className="form-control inputSearch"
                          name="inputName"
                          onChange={[Function]}
                          placeholder="Search business by name ..."
                          type="text"
                        />
                        <div
                          className="input-group-append"
                        >
                          <button
                            className="btn btn-outline-secondary btnSearch"
                            onClick={[Function]}
                            type="button"
                          >
                            Search
                          </button>
                        </div>
                      </div>
                    </div>
                  </Col>
                </Row>
              </TabPane>
              <TabPane
                tabId="location"
                tag="div"
              >
                <Row
                  tag="div"
                >
                  <Col
                    sm="12"
                    tag="div"
                    widths={
                      Array [
                        "xs",
                        "sm",
                        "md",
                        "lg",
                        "xl",
                      ]
                    }
                  >
                    <div
                      aria-labelledby="pills-name-tab"
                      className="tab-pane fade show active"
                      id="pills-name"
                      role="tabpanel"
                    >
                      <div
                        className="input-group mb-3"
                      >
                        <input
                          aria-describedby="basic-addon2"
                          aria-label="pills-name-input"
                          className="form-control inputSearch"
                          name="inputLocation"
                          onChange={[Function]}
                          placeholder="Search business by location ..."
                          type="text"
                        />
                        <div
                          className="input-group-append"
                        >
                          <button
                            className="btn btn-outline-secondary btnSearch"
                            onClick={[Function]}
                            type="button"
                          >
                            Search
                          </button>
                        </div>
                      </div>
                    </div>
                  </Col>
                </Row>
              </TabPane>
              <TabPane
                tabId="category"
                tag="div"
              >
                <Row
                  tag="div"
                >
                  <Col
                    sm="12"
                    tag="div"
                    widths={
                      Array [
                        "xs",
                        "sm",
                        "md",
                        "lg",
                        "xl",
                      ]
                    }
                  >
                    <div
                      aria-labelledby="pills-name-tab"
                      className="tab-pane fade show active"
                      id="pills-name"
                      role="tabpanel"
                    >
                      <div
                        className="input-group mb-3"
                      >
                        <input
                          aria-describedby="basic-addon2"
                          aria-label="pills-name-input"
                          className="form-control inputSearch"
                          name="inputCategory"
                          onChange={[Function]}
                          placeholder="Search business by category ..."
                          type="text"
                        />
                        <div
                          className="input-group-append"
                        >
                          <button
                            className="btn btn-outline-secondary btnSearch"
                            onClick={[Function]}
                            type="button"
                          >
                            Search
                          </button>
                        </div>
                      </div>
                    </div>
                  </Col>
                </Row>
              </TabPane>
            </TabContent>
          </div>
        </div>
      </div>
    </div>
  </section>
  <Footer />
</div>
`;

exports[`Page404 renders correctly Succeeds 1`] = `
<div>
  <section
    id="body"
  >
    <NavBar />
    <div
      className="container image404 col-min-6"
      id="loginContainer"
    >
      <div
        className="wrapper"
      >
        <div
          className="card"
        >
          <div
            className="row"
          >
            <div
              className="col-sm-4"
            >
              <img
                alt="page404"
                className="img-fluid"
                id="img404"
                src="chimp404.jpg"
              />
            </div>
            <div
              className="col-sm-8"
            >
              <div
                className="align-middle"
                id="text404"
              >
                <h1>
                  404
                </h1>
                <h3>
                  Page Not Found
                </h3>
                <h4>
                  The page you are looking for could be found on this server.
                </h4>
                <br />
                <a
                  className="btn btn-primary"
                  href="/"
                >
                  Take Me Home
                </a>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </section>
  <Footer />
</div>
`;
